packages/checkout/sdk/src/sdk.ts

```diff
@@ -119,22 +121,40 @@ export class Checkout {
     // Preload the configurations
     await checkout.config.remote.getConfig();
 
+    try {
+      const factory = await this.loadEsModules(init.config, init.version);
+      return factory;
+    } catch (err: any) {
+      throw new CheckoutError(
+        'Failed to load widgets script',
+        CheckoutErrorType.WIDGETS_SCRIPT_LOAD_ERROR,
+        { error: err },
+      );
+    }
+  }
+
+  private async loadUmdBundle(
+    config: WidgetConfiguration,
+    version?: SemanticVersion,
+  ) {
+    const checkout = this;
+
     const factory = new Promise<ImmutableCheckoutWidgets.WidgetsFactory>((resolve, reject) => {
       function checkForWidgetsBundleLoaded() {
```

Comment:
One thing I think worth introducing here is `bundleLoadAttempts` as an incremental number. Each time `checkForWidgetsBundleLoaded` is called and we can increment that.. after x number of attempts we throw an error and bail out? Just so we don't get stuck forever..

Reply:
Yeah good idea - I'll add something in as part of this PR to break out after ~10 attempts or so

Reply:
Hm looks like on slower networks this can run for a bit so its hard to actually put a number here on attempts (took like ~60 attempts on 'fast 3g').

Reply:
Yep sounds good, timings like this are always hard.

Reply:
Hey @imx-mikhala , one change I'd love to make here that we didn't do last time was adding the script onload event listener to resolve the promise that way. I did some testing on this and it worked correctly.

Reply:
The problem with this last time is that we were only creating the event listener after the script had loaded. We need to create the script element. Hook into the on load event and resolve the promise from in there. And then add script.src which will start loading it.